.github/workflows/release.yml

```
@@ -0,0 +1,41 @@
+name: Release
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+    tags:
+      - '**'  # allow every tag including hierarchical tags like v1.0/beta
```

`staging` branch should be released to TestPypi
```suggestion
    branches:
      - staging
```